docs/datastore/cloudsql_postgres.md

```diff
@@ -0,0 +1,156 @@
+# Setup and configure Cloud SQL for Postgres
+
+## Before you begin
+
+1. Make sure you have a Google Cloud project and billing is enabled.
+
+1. Set your `PROJECT_ID` environment variable:
+
+    ```bash
+    export PROJECT_ID=<YOUR_PROJECT_ID>
+    ```
+
+1. [Install](https://cloud.google.com/sdk/docs/install) the gcloud CLI.
+
+1. Set gcloud project:
+
+    ```bash
+    gcloud config set project $PROJECT_ID
+    ```
+
+1. Enable APIs:
+
+    ```bash
+    gcloud services enable sqladmin.googleapis.com \
+                           aiplatform.googleapis.com
+    ```
+
+1. [Install python][install-python] and set up a python [virtual environment][venv].
+
+1. Make sure you have python version 3.10+ installed.
+
+    ```bash
+    python -V
+    ```
+
+1. Download and install [postgres-client cli (`psql`)][install-psql].
+
+1. Install the [Cloud SQL Auth Proxy client][install-cloudsql-proxy].
+
+[install-python]: https://cloud.google.com/python/docs/setup#installing_python
+[venv]: https://cloud.google.com/python/docs/setup#installing_and_using_virtualenv
+[install-psql]: https://www.timescale.com/blog/how-to-install-psql-on-mac-ubuntu-debian-windows/
+[install-cloudsql-proxy]: https://cloud.google.com/sql/docs/postgres/connect-instance-auth-proxy#install-proxy
+
+
+## Create a Cloud SQL for PostgreSQL instance
+
+1. Set environment variables. For security reasons, use a different password for
+   `$DB_PASS` and note it for future use:
+
+    ```bash
+    export DB_PASS=my-cloudsql-pass
+    export DB_USER=postgres
+    export INSTANCE=my-cloudsql-instance
+    export REGION=us-central1
+    ```
+
+1. Create a PostgreSQL instance:
+
+    ```bash
+    gcloud sql instances create $INSTANCE \
+        --database-version=POSTGRES_14 \
+        --cpu=4 \
+        --memory=16GB \
+        --region=$REGION
+    ```
+
+1. Set password for postgres user:
+
+    ```bash
+    gcloud sql users set-password $DB_USER \
+        --instance=$INSTANCE \
+        --password=$DB_PASS
+    ```
+
+
+## Connect to the Cloud SQL instance
+
+1. Set environment variables:
+
+    ```bash
+    export INSTANCE_CONNECTION_NAME=$PROJECT_ID:$REGION:$INSTANCE
+    ```
+
+1. Connect to instance using cloud sql proxy:
+
+    ```bash
+    ./cloud-sql-proxy $INSTANCE_CONNECTION_NAME
+    ```
```

Comment:
nit: should we just combine these into a single command?

Also -- should $PROJECT_ID be set above? 

```suggestion
1. Connect to instance using cloud sql proxy:

    ```bash
    ./cloud-sql-proxy $PROJECT_ID:$REGION:$INSTANCE
    ```
```

Reply:
The $PROJECT_ID is set at the before you begin section since it is usually shared across multiple sections. And to keep consistency with other datastore's instructions, I kept it at the same section above. Let me know if it makes more sense to move it down. Thank you!